### PR TITLE
Components update

### PR DESCRIPTION
**Pull request by SotoiGhost**
This PR includes updates for:
* Firebase Core v6.6.0 (a.k.a 6.15.0)
* Firebase InstanceID v4.3.0 (a.k.a 6.15.0)
* Firebase Analytics v6.2.1 (a.k.a 6.15.0)
* Firebase AdMob v7.53.1 (a.k.a 6.15.0)
* Google SignIn v5.0.2
* Firebase Auth v6.4.2 (a.k.a 6.15.0)
* Firebase Crashlytics v3.14.0.2
* Google Cast v4.4.6
* Google Maps v3.7.0
* Google Places v3.7.0
* Firebase ABTesting v3.1.2
* Firebase RemoteConfig v4.4.6
* Firebase Performance Monitoring v3.1.9 (a.k.a 6.15.0)

Also, replaced `RestoreAssemblyResource` with `BundleResource` and `NativeReference`

### CODE REVIEW

File: source/Firebase/Auth/ApiDefinition.cs [hunk 1]
```diff
@@ -601,13 +601,23 @@ interface OAuthProvider : FederatedAuthProvider {
 
 		// +(FIROAuthCredential * _Nonnull)credentialWithProviderID:(NSString * _Nonnull)providerID IDToken:(NSString * _Nonnull)IDToken accessToken:(NSString * _Nullable)accessToken;
 		[Static]
-		[Export ("credentialWithProviderID:IDToken:accessToken:")]
+		[Export("credentialWithProviderID:IDToken:accessToken:")]
```
Comment (dalexsoto) [inline source/Firebase/Auth/ApiDefinition.cs]:
```suggestion
		[Export ("credentialWithProviderID:IDToken:accessToken:")]
```

File: source/Firebase/Auth/ApiDefinition.cs [hunk 2]
```diff
@@ -748,6 +758,12 @@ interface User : UserInfo
 		[Export ("reauthenticateAndRetrieveDataWithCredential:completion:")]
 		void ReauthenticateAndRetrieveData (AuthCredential credential, [NullAllowed] AuthDataResultHandler completion);
 
+		// -(void)reauthenticateWithProvider:(id<FIRFederatedAuthProvider> _Nonnull)provider UIDelegate:(id<FIRAuthUIDelegate> _Nullable)UIDelegate completion:(FIRAuthDataResultCallback _Nullable)completion __attribute__((swift_name("reauthenticate(with:uiDelegate:completion:)"))) __attribute__((availability(ios, introduced=8.0)));
+		[iOS (8,0)]
+		[Async]
+		[Export ("reauthenticateWithProvider:UIDelegate:completion:")]
+		void Reauthenticate (FederatedAuthProvider provider, [NullAllowed] AuthUIDelegate UIDelegate, [NullAllowed] AuthDataResultHandler completion);
```
Comment (dalexsoto) [inline source/Firebase/Auth/ApiDefinition.cs]:
```suggestion
		void Reauthenticate (FederatedAuthProvider provider, [NullAllowed] AuthUIDelegate @delegate, [NullAllowed] AuthDataResultHandler completion);
```

File: source/Firebase/Auth/ApiDefinition.cs [hunk 3]
```diff
@@ -779,6 +795,12 @@ interface User : UserInfo
 		[Export ("linkWithCredential:completion:")]
 		void Link (AuthCredential credential, [NullAllowed] AuthDataResultHandler completion);
 
+		// -(void)linkWithProvider:(id<FIRFederatedAuthProvider> _Nonnull)provider UIDelegate:(id<FIRAuthUIDelegate> _Nullable)UIDelegate completion:(FIRAuthDataResultCallback _Nullable)completion __attribute__((swift_name("link(with:uiDelegate:completion:)"))) __attribute__((availability(ios, introduced=8.0)));
+		[iOS (8,0)]
+		[Async]
+		[Export ("linkWithProvider:UIDelegate:completion:")]
+		void Link (FederatedAuthProvider provider, [NullAllowed] AuthUIDelegate UIDelegate, [NullAllowed] AuthDataResultHandler completion);
```
Comment (dalexsoto) [inline source/Firebase/Auth/ApiDefinition.cs]:
```suggestion
		void Link (FederatedAuthProvider provider, [NullAllowed] AuthUIDelegate @delegate, [NullAllowed] AuthDataResultHandler completion);
```

File: source/Google/MobileAds/ApiDefinition.cs [hunk 4]
```diff
@@ -432,14 +471,85 @@ interface Request : INSCopying {
...
 	[BaseType (typeof (NSError), Name = "GADRequestError")]
 	interface RequestError {
+		// extern NSString *const kGADErrorDomain __attribute__((visibility("default")));
+		[Internal]
+		[Field ("kGADErrorDomain", "__Internal")]
+		NSString _ErrorDomain { get; }
+
+		[Static]
+		[Wrap ("_ErrorDomain.ToString ()")]
+		string ErrorDomain { get; }
```
Comment (dalexsoto) [inline source/Google/MobileAds/ApiDefinition.cs]: Why not just? Error domain NSStrings do not make much sense as `string`

```
[Field ("kGADErrorDomain", "__Internal")]
NSString ErrorDomain { get; }
```

File: source/Google/MobileAds/Enums.cs [hunk 5]
```diff
@@ -38,51 +38,57 @@ public enum AdLoaderAdType
...
+		[Obsolete ("Use AdAlreadyUsed enum value instead.")]
+		InterstitialAlreadyUsed = 6,
+		MediationDataError = 7,
+		MediationAdapterError = 8,
+		MediationInvalidAdSize = 10,
+		InternalError = 11,
+		InvalidArgument = 12,
+		ReceivedInvalidResponse = 13,
+		[Obsolete("Use AdAlreadyUsed enum value instead.")]
+		RewardedAdAlreadyUsed = 14,
```
Comment (dalexsoto) [inline source/Google/MobileAds/Enums.cs]: You can go the extra mile and do this instead 
```suggestion
		RewardedAdAlreadyUsed = AdAlreadyUsed,
```

Same for the other obsoletes

File: source/Google/MobileAds/ApiDefinition.cs (hunk 4 above)
Comment (SotoiGhost) [reply source/Google/MobileAds/ApiDefinition.cs]: @dalexsoto `ErrorDomain` was bound as `string`, changing this to `NSString` could cause breaking changes.

**Pull request merged**